keps/sig-scheduling/5055-dra-device-taints-and-tolerations/README.md

```diff
@@ -258,30 +306,85 @@ Device and node taints are applied independently. A node taint applies to all
 pods on a node, whereas a device taint affects claim allocation and only those
 pods using the claim.
 
+### kubectl describe DeviceTaintRule
+
+An extension of `kubectl describe` for DeviceTaintRule uses the same code as
+the eviction controller to calculate which devices are matched and which pods
+would need to get evicted. The input for that code is the single
+DeviceTaintRule which gets described and the complete set of ResourceSlices,
+ResourceClaims and Pods.
+
+Calculating the outcome on the client side was chosen because the alternative,
+always updating the status of DeviceTaintRules on the server side, would cause
+permanent churn regardless whether someone wants the information or not.
+
+The exact output will be decided during the implementation phase.
+It can change at any time because the `kubectl describe` output does not fall
+under any Kubernetes API stability guarantees.
+
 ### API
 
-The ResourceSlice content gets extended:
+The ResourceSlice content gets extended. A natural place for a taint that
+affects one specific device would be inside the `Device` struct. The problem
+with that approach is that with 128 allowed devices per ResourceSlice the
+maximum size of each taint would have to be very small to keep the worst-case
+size of a ResourceSlice within the limits imposed by etcd and the API request
+machinery.
+
+Therefore a different approach is used where each ResourceSlice either provides
+information about devices or about taints, but never both. Using the
+ResourceSlice for taints instead of a different type has the advantage that the
+existing mechanisms and code for publishing and consuming the information can
+be reused. The same approach is likely to be used for other additional
+ResourceSlice pool information, like mixin definitions. The downside is that at
+least two ResourceSlices are necessary once taints get published by a DRA
+driver.
 
 ```Go
-type Device struct {
+type ResourceSliceSpec struct {
```

Comment:
I'd add the nessessary validation details for the new fields:

- Enforce mutual exclusivity of Devices vs Taints with schema validation (oneOf).
- Validate max 4 taints per device and max 32 per slice server-side.
- For Description/Data, specify truncation vs rejection for over-length.